src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,217 @@
+#include "HexagonAlignLoads.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Scope.h"
+#include "Bounds.h"
+#include "ModulusRemainder.h"
+#include "Simplify.h"
+namespace Halide {
+namespace Internal {
+using std::vector;
+
+class HexagonAlignLoads : public IRMutator {
```

Comment:
This logic is not specific to Hexagon. There may be other platforms where it makes sense in the future. I'd remove the name "Hexagon" from the pass.